x/pairing/keeper/scores/pairing_slot.go

```diff
@@ -2,20 +2,37 @@ package scores
 
 // PairingSlot holds the set of requirements for a slot.
 // It also holds the number of identical slots required for the pairing (count)
-type PairingSlot struct {
-	Reqs  map[string]ScoreReq
-	Count int
+
+type PairingSlotInf interface {
+	Requirements() map[string]ScoreReq
+}
+
+type PairingSlotGroup struct {
+	Reqs    map[string]ScoreReq
+	indexes []int
+}
+
+func (s PairingSlotGroup) Indexes() []int {
+	return s.indexes
+}
+
+func (s PairingSlotGroup) Requirements() map[string]ScoreReq {
+	return s.Reqs
+}
+
+func NewPairingSlotGroup(pairingSlot *PairingSlot) *PairingSlotGroup {
+	return &PairingSlotGroup{indexes: []int{pairingSlot.Index}, Reqs: pairingSlot.Reqs}
 }
 
-func NewPairingSlot() *PairingSlot {
-	return &PairingSlot{Count: 1}
+func (psg *PairingSlotGroup) AddToGroup(pairingSlot *PairingSlot) {
+	psg.indexes = append(psg.indexes, pairingSlot.Index)
```

Comment:
verify that they have the same requirements?

Reply:
the only use of this is after checking equal no reason to check twice